File: bot.py
# bot.py
import os
import sys
import importlib
from pathlib import Path
from dotenv import load_dotenv
from telegram import Update, Bot
from telegram.ext import ApplicationBuilder, CommandHandler, MessageHandler, filters, CallbackQueryHandler, ContextTypes

# ======================
# إعداد البيئة
# ======================
load_dotenv()
BOT_TOKEN = os.getenv("BOT_TOKEN")
if not BOT_TOKEN:
    raise Exception("❌ BOT_TOKEN غير موجود")

# ======================
# المطورين
# ======================
DEV_IDS = [5860391324, 7076215547, 7855813063]  # ضع هنا كل المطورين

# ======================
# قاعدة البيانات
# ======================
import db_manager
db_manager.init_db() if hasattr(db_manager, "init_db") else None

# ======================
# تحميل الموديولات ديناميكياً
# ======================
cmd_modules = {}
game_modules = {}
module_errors = {}

def load_modules():
    global cmd_modules, game_modules, module_errors
    cmd_modules.clear()
    game_modules.clear()
    module_errors.clear()

    base_path = Path(__file__).parent
    print("📦 جاري تحميل الموديولات...")

    for file in base_path.glob("*.py"):
        if file.name in ["bot.py", "db_manager.py"]:
            continue
        module_name = file.stem
        try:
            spec = importlib.import_module(module_name)
            importlib.reload(spec)
            if hasattr(spec, "handle"):
                if module_name.startswith("cmd_"):
                    cmd_modules[module_name] = spec
                elif module_name.startswith("game_"):
                    game_modules[module_name] = spec
            print(f"✅ تم تحميل: {module_name}")
        except Exception as e:
            module_errors[module_name] = str(e)
            print(f"⚠️ خطأ تحميل {module_name}: {e}")
            for dev_id in DEV_IDS:
                try:
                    bot = Bot(BOT_TOKEN)
                    bot.send_message(dev_id, f"⚠️ خطأ تحميل:\n{module_name}\n{str(e)}")
                except:
                    pass

load_modules()

# ======================
# أدوات مساعدة
# ======================
def is_developer(uid):
    return uid in DEV_IDS

# ======================
# أوامر أساسية
# ======================
async def start(update: Update, context: ContextTypes.DEFAULT_TYPE):
    await update.message.reply_text("👑 البوت شغال وبكامل الهيبة.")

async def update_modules(update: Update, context: ContextTypes.DEFAULT_TYPE):
    uid = update.effective_user.id
    if not is_developer(uid):
        return await update.message.reply_text("❌ هذا الأمر للمطورين فقط")
    load_modules()
    reply = "🔄 تم تحديث الموديولات\n\n✅ CMD:\n"
    reply += "\n".join(cmd_modules.keys()) or "—"
    reply += "\n\n🎮 GAME:\n"
    reply += "\n".join(game_modules.keys()) or "—"
    if module_errors:
        reply += "\n\n⚠️ أخطاء:\n"
        for f, e in module_errors.items():
            reply += f"• {f}: {e}\n"
    await update.message.reply_text(reply)

async def shutdown(update: Update, context: ContextTypes.DEFAULT_TYPE):
    uid = update.effective_user.id
    if is_developer(uid):
        await update.message.reply_text("🛑 يتم إيقاف البوت...")
        sys.exit(0)

async def restart(update: Update, context: ContextTypes.DEFAULT_TYPE):
    uid = update.effective_user.id
    if is_developer(uid):
        await update.message.reply_text("♻️ إعادة تشغيل...")
        os.execv(sys.executable, ["python"] + sys.argv)

# ======================
# التعامل مع الرسائل النصية
# ======================
async def handle_text(update: Update, context: ContextTypes.DEFAULT_TYPE):
    for module in cmd_modules.values():
        if hasattr(module, "handle"):
            try:
                await module.handle(update, context, db_manager, DEV_IDS)
            except Exception as e:
                print("⚠️ خطأ CMD:", e)
    for module in game_modules.values():
        if hasattr(module, "handle"):
            try:
                await module.handle(update, context, db_manager, DEV_IDS)
            except Exception as e:
                print("⚠️ خطأ GAME:", e)

# ======================
# التعامل مع أزرار Inline
# ======================
async def handle_callback(update: Update, context: ContextTypes.DEFAULT_TYPE):
    for module in cmd_modules.values():
        if hasattr(module, "handle_callback"):
            try:
                await module.handle_callback(update, context, db_manager, DEV_IDS)
            except Exception as e:
                print("⚠️ خطأ Callback:", e)

# ======================
# تشغيل البوت
# ======================
app = ApplicationBuilder().token(BOT_TOKEN).build()
app.add_handler(CommandHandler("start", start))
app.add_handler(CommandHandler("update_modules", update_modules))
app.add_handler(CommandHandler("shutdown", shutdown))
app.add_handler(CommandHandler("restart", restart))
app.add_handler(MessageHandler(filters.TEXT & (~filters.COMMAND), handle_text))
app.add_handler(CallbackQueryHandler(handle_callback))

print("🚀 البوت شغال وبكامل الهيبة")
app.run_polling()

File: bot.js
// bot.js
require('dotenv').config();
const { Telegraf } = require('telegraf');
const fs = require('fs');
const path = require('path');

const BOT_TOKEN = process.env.BOT_TOKEN;
const DEV_IDS = process.env.DEV_IDS ? process.env.DEV_IDS.split(',').map(id => id.trim()) : [];

if (!BOT_TOKEN) throw new Error('❌ BOT_TOKEN غير موجود');

const bot = new Telegraf(BOT_TOKEN);

// ======================
// الموديولات
// ======================
const cmdModules = {};
const gameModules = {};
const moduleErrors = {};

function loadModules() {
    cmdModules = {};
    gameModules = {};
    moduleErrors = {};

    const basePath = __dirname;
    console.log('📦 جاري تحميل الموديولات...');

    fs.readdirSync(basePath).forEach(file => {
        if (!file.endsWith('.js') || file === 'bot.js') return;

        const moduleName = file.replace('.js', '');
        try {
            const mod = require(path.join(basePath, file));
            if (mod.handle) {
                if (file.startsWith('cmd_')) cmdModules[moduleName] = mod;
                if (file.startsWith('game_')) gameModules[moduleName] = mod;
            }
            console.log(`✅ تم تحميل الموديول: ${moduleName}`);
        } catch (err) {
            moduleErrors[file] = err.message;
            console.error(`⚠️ خطأ تحميل الموديول ${file}:\n`, err);
            DEV_IDS.forEach(id => {
                bot.telegram.sendMessage(id, `⚠️ خطأ تحميل:\n${file}\n${err.message}`).catch(() => {});
            });
        }
    });
}

loadModules();

// ======================
// أدوات مساعدة
// ======================
async function isAdmin(ctx, userId) {
    try {
        const member = await ctx.getChatMember(userId);
        return ['administrator', 'creator'].includes(member.status);
    } catch {
        return false;
    }
}

// ======================
// /start
// ======================
bot.start(ctx => {
    console.log(`📩 أمر /start من: ${ctx.from.id}`);
    ctx.reply('👑 البوت شغال.');
});

// ======================
// التعامل مع الرسائل
// ======================
bot.on('text', async ctx => {
    const uid = ctx.from.id;
    const chatId = ctx.chat.id;
    const text = ctx.message.text.trim();

    console.log(`📩 رسالة من ${uid} في ${chatId}: ${text}`);

    // أمر تحديث الموديولات
    if (text.toLowerCase() === 'تحديث' && DEV_IDS.includes(uid)) {
        loadModules();
        let reply = '🔄 تم تحديث الموديولات\n\n✅ CMD:\n';
        reply += Object.keys(cmdModules).join('\n') + '\n\n🎮 GAME:\n';
        reply += Object.keys(gameModules).join('\n');

        if (Object.keys(moduleErrors).length > 0) {
            reply += '\n\n⚠️ أخطاء:\n';
            for (const [fname, err] of Object.entries(moduleErrors)) {
                reply += `• ${fname}: ${err}\n`;
            }
        }

        ctx.reply(reply);
        return;
    }

    // أمر ريست البوت
    if (['ريست', 'إعادة تشغيل'].includes(text) && DEV_IDS.includes(uid)) {
        ctx.reply('♻️ يتم إعادة تشغيل البوت...');
        console.log('♻️ إعادة تشغيل البوت...');
        process.exit(0);
    }

    // تمرير الرسائل لبقية الموديولات
    for (const [moduleName, module] of Object.entries(cmdModules)) {
        try {
            module.handle(bot, ctx);
        } catch (err) {
            console.error(`⚠️ خطأ في موديول CMD ${moduleName}:\n`, err);
            DEV_IDS.forEach(id => {
                bot.telegram.sendMessage(id, `⚠️ خطأ في تنفيذ CMD ${moduleName}:\n${err.message}`).catch(() => {});
            });
        }
    }

    for (const [moduleName, module] of Object.entries(gameModules)) {
        try {
            module.handle(bot, ctx);
        } catch (err) {
            console.error(`⚠️ خطأ في موديول GAME ${moduleName}:\n`, err);
            DEV_IDS.forEach(id => {
                bot.telegram.sendMessage(id, `⚠️ خطأ في تنفيذ GAME ${moduleName}:\n${err.message}`).catch(() => {});
            });
        }
    }
});

// ======================
// تشغيل البوت
// ======================
bot.launch().then(() => console.log('🚀 البوت شغال، جاري الاستماع للرسائل...'));
